x/ibc/04-channel/keeper/packet.go

```diff
@@ -174,11 +179,12 @@ func (k Keeper) RecvPacket(
 
 	// check if packet timeouted by comparing it with the latest height of the chain
 	if packet.GetTimeoutHeight() != 0 && uint64(ctx.BlockHeight()) >= packet.GetTimeoutHeight() {
-		return nil, types.ErrPacketTimeout
+		return nil, sdkerrors.Wrapf(types.ErrPacketTimeout, "packet timeout height (%d) has been passed by the current block height (%d)", packet.GetTimeoutHeight(), uint64(ctx.BlockHeight()))
 	}
 
-	if packet.GetTimeoutTimestamp() != 0 && ctx.BlockTime().UnixNano() >= packet.GetTimeoutTimestamp() {
-		return nil, types.ErrPacketTimeout
+	// check if packet timeouted by comparing it with the latest timestamp of the chain
+	if packet.GetTimeoutTimestamp() != 0 && uint64(ctx.BlockTime().UnixNano()) >= packet.GetTimeoutTimestamp() {
```

Comment:
if the block time is negative, the conversion to uint64 will panic. Should I add any check for this or maybe just leave a note?

Reply:
it's ok imo